quickstart/101-application-gateway/main.tf

```diff
@@ -0,0 +1,176 @@
+terraform {
+
+  required_version = ">=0.12"
+  
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">=2.97.0"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "rg1" {
```

Comment:
Should make use of the random_pet resource to abstract names like this. Please refer to this PR for more context: https://github.com/Azure/terraform/pull/81/files